### DISCUSSION TITLE
Nested Queries Limitation

### DISCUSSION THREAD

**Discussion by VitalijF** [Netflix/dgs-framework#1165] [General]
if there is a way to limit the max amount of nested queries?

**Comment by srinivasankavitha** [comment]
there is a MaxQueryDepthInstrumentation that you can use for this in graphql-java: https://www.graphql-java.com/documentation/v17/instrumentation/#query-depth-instrumentation

Should work seamlessly with the DGS framework.

**Reply by VitalijF** [reply]
@srinivasankavitha Thank you. Additional question. Is it possible to limit it base on response
Example:
```
{
   baseQuery {
      id
      nestedQueryA {
         id
         nestedQueryB {
          id
}
}
}
```
So, if we receive 100 elements for base query set limit only for 20 nested query objects

**Comment by srinivasankavitha** [comment]
You will need to model that in the schema, i.e. have an input parameter to specify how many results are returned if you need that capability.

**Reply by VitalijF** [reply]
@srinivasankavitha Is it possible to take number of elements which have been already fetched(from dfe or somewhere)?
